gammapy/modeling/models/spatial.py

```diff
@@ -1376,7 +1377,7 @@ def evaluate_geom(self, geom):
 
         """
         coords = geom.get_coord(frame=self.frame, sparse=True)
-        return self(coords.lon, coords.lat)
+        return self(Angle(coords.lon).wrap_at(180 * u.deg), coords.lat)
```

Comment:
this fix should be applied to the whitin evaluate function adding lon=Angle(lon).wrap_at(self._wrap_angle)`

Reply:
Do you mean inside the `evaluate_geom` function? As far as I understand, this is an issue with the geom not the evaluation?

Reply:
`evaluate_geom ` call `evaluate` and the problem already appears here if the given lon is not valid
https://github.com/gammapy/gammapy/blob/fc886b94357e131de35787657c7592f7d42f232e/gammapy/modeling/models/spatial.py#L1352C9-L1352C17

Reply:
Oh yes, sorry. 
I have fixed it. Should I update the test to be at the GC, or leave it as it is?

Reply:
yes you can update the existing test to be at the GC

Reply:
I added tests at galactic center an anticenter and I don't see any fails, same trying the doc example in these regions.
Do you have an simple example to reproduce the fail ?

Reply:
The issue occurs when you chose the GC: 
`geom = WcsGeom.create(skydir=(0, 0), npix=(2, 2), binsz=0.3, frame="galactic")`

Then `new_model.evaluate_geom(geom.to_image()) = [[NaN, 3], [NaN, 2]]`

Reply:
ok the definition of the `wrap_angle` was not working if the wrap angle was not 0 in the input coords, now it should work for all longitude and I added a test that loop over -360<l<360